Newcomers_Guide: Improves emphasis of material

kriti21 opened the pull request:
This adds color schemes for different types of blocks
and catgorizes them into important, caution and notes.

Closes https://github.com/coala/coala/issues/3704

newbazz commented on docs/_static/custom.css:
```diff
@@ -81,6 +81,30 @@ div.admonition, div.note {  /* Ignore CSSLintBear */
+    color: white;
+    font-family: Roboto;
+    padding: 2em;
+    position: relative;
```
Relative positioning isnt default one?

newbazz commented on docs/_static/custom.css:
```diff
@@ -81,6 +81,30 @@ div.admonition, div.note {  /* Ignore CSSLintBear */
+    color: white;
+    font-family: Roboto;
+    padding: 2em;
+    position: relative;
```
here as well

newbazz commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -208,27 +208,27 @@ Now you can start working on it!
+        * You shouldn't close any PR, instead mark them as work in progress
 
-            corobo mark wip <full url>
+            **corobo mark wip <full url>**
```
I ud suggest boldning in a different commit <if required>, what you say @Makman2 ?

adhikasp replied:
I think that is unnecessary. This could be considered as part of emphasizing and relevant to the commit.

newbazz replied:
@adhikasp (Y)

Makman2 replied:
Hm this should be rendered as code, so it should be kept. Imo a different issue.

kriti21 replied:
@Makman2 These colons are not for text but to use a different block defined in css. If i keep these colons, the background will be as defined here https://github.com/coala/coala/blob/1db00d706353bb0d8a175c6d561df21e51cad6e7/docs/_static/custom.css#L108 or else the left padding is disturbed.
And that is why, the rendered code also needs to be bold. Additionally, bold text appears better over yellow background. But this i can keep normal for this PR and open a separate issue to bold this text. What do you suggest?

ishanSrt commented:
does this change the content of the link: http://api.coala.io/en/latest/Developers/Newcomers_Guide.html or just for https://github.com/coala/coala/blob/master/docs/Developers/Newcomers_Guide.rst or are the two same?

kriti21 commented:
@ishanSrt both of the above links you mentioned refer to the same thing but the changes I have made are for http://api.coala.io/en/latest/Developers/Newcomers_Guide.html Because this is sort of a design issue, major change was on css rendering part.

Makman2 commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -310,13 +308,11 @@ into your bash prompt. This will analyze your code and help you fix it.
    Before committing your changes, please check that you are indeed in a
    development branch created in step 3. To check if you are in a branch, type:
 
-   ::
```
different issue :)

Makman2 commented on docs/_static/custom.css:
```diff
@@ -81,6 +81,14 @@ div.admonition, div.note {  /* Ignore CSSLintBear */
+
+div.important {  /* Ignore CSSLintBear */
+    background-color: #783238;
+}
```
Isn't this maybe a bit too dark?

Btw I would like to see a screenshot how it looks like when it's rendered :)

jayvdb replied:
We should netlify the docs in this repo.
I had a WIP for that, but it was blocked by https://github.com/coala/devops/issues/90

kriti21 replied:
I have added a screenshot in the comments anyway. Is that okay or any changes required ? @jayvdb

Makman2 commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -175,7 +175,7 @@ Now it is time to pick an issue.
 You can view `all Newcomer issues on GitHub <https://coala.io/new>`_.
 
-.. note::
+.. caution::
```
Maybe this is "important"?

We should definitely clear what messages are "caution" and what "important"

Makman2 commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -208,27 +208,27 @@ Now you can start working on it!
     You can do amazing stuff using corobo.
 
         * Issue assigning as stated earlier.
-        * File issues::
```
Different issue removing the colons :)

jayvdb replied:
@kriti21 has an issue been created about this yet?
If not, please create one.  if you describe clearly what needs to be done, and show them this PR, it is a newcomer issue, and we'll find a newcomer to do it, and you need to remove this stuff from your PR.

kriti21 replied:
But adding the extra colon makes the background of all the following corobo commands grey colored. Is is okay to push a PR like that (not looking good) ? @jayvdb

jayvdb replied:
Create a separate issue.  The extra colon is a syntax *bug*, and is unrelated to your issue.

Makman2 commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -208,27 +208,27 @@ Now you can start working on it!
+            **corobo mark wip <full url>**
 
-        * To label a PR as pending review::
+        * To label a PR as pending review
```
Different issue

Makman2 commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -208,27 +208,27 @@ Now you can start working on it!
+        * To see all of the corobo commands
 
-            corobo help
+            **corobo help**
```
Different issue

Makman2 commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -208,27 +208,27 @@ Now you can start working on it!
     link: `Writing good commits <http://coala.io/commit>`_.
 
-.. seealso::
+.. caution::
```
Rather "note" or "important"

Makman2 commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -449,7 +445,7 @@ Now you should `tell us about your experience <https://coala.io/newform>`_ and
 
 
-.. note::
+.. caution::
```
Maybe we want a new tag "critical" or so, because this is so damn important imo 😅

kriti21 replied:
I will add critical ;) Is that okay ?

kriti21 replied:
@Makman2 I am using tag "critical" for this one. Can you suggest me the background color to be used with this? Red is already used for important and yellow for caution.

Makman2 commented:
unack 80ead210ddcea628751cb015bcc2b678ee908ee3

kriti21 commented:
![colorbrightness](https://user-images.githubusercontent.com/19872082/37017505-dbf9657e-2136-11e8-8c03-6444b2f6b84d.png)

kriti21 commented:
Also "critical" is not supported in the Specific Admonitions. I have used "Attention" instead. Here are all the available options http://docutils.sourceforge.net/docs/ref/rst/directives.html#specific-admonitions , if you could suggest anything better. Or I could make a generic admonition if that would be fine because we do not really need critical tag that often.

kriti21 commented:
@HizkiFW I have added `/* Ignore CSSLintBear */ ` to avoid https://github.com/coala/coala/pull/5119#pullrequestreview-110256850 error. Using only `.important` or `.caution` will not work as these are used in other developer guides. I have changed the colors though. Please review. :)

HizkiFW commented:
Regarding the CSS, I found out that using the class names alone doesn't work because the theme (`alabaster.css`) and `basic.css` both have the element name on their style definitions, thus overriding the class-only definitions.

This could be fixed by removing the `div`s from those files, but they're part of another project (https://github.com/bitprophet/alabaster). Maybe we could make a fork of the theme and use that instead.

Also, @kriti21:
> Using only `.important` or `.caution` will not work as these are used in other developer guides.

what do you mean by "other developer guides"? Which ones?

kriti21 commented:
>what do you mean by "other developer guides"? Which ones?
I just meant that is will not work that way i.e. using `.important` only. Said something wrong there. Please neglect that @HizkiFW 
`custom.css` is a common file used for all the guides here https://github.com/coala/coala/tree/master/docs/Developers . Making any changes might have more impacts than required. What do you suggest I should do?

HizkiFW commented:
You're right, changing that is definitely going to break more than just a few things. Just leave it like that but also add a comment with a link to this issue.

Phantom-42 commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -281,8 +281,6 @@ fork repository.
    In order to avoid the above mentioned cases you can create a new branch
    where you will work on the issue. To do that run:
 
-   ::
```
This fix is unrelated to improving emphasis of material. I think it is better to have this in a separate change.

jayvdb commented on docs/_static/custom.css:
```diff
@@ -81,6 +81,18 @@ div.admonition, div.note {  /* Ignore CSSLintBear */
     transition: box-shadow .25s;
 }
 
+div.attention {  /* Ignore CSSLintBear */
```
I think we can use `!important` to override other stylesheets.
@blazeu ?

Ignoring the bear is not a good option, as it means all of its rules are ignored.

blazeu replied:
Yeah sure, and `div.attention` is a bit excessive, no need to put the element name unless necessary

kriti21 commented:
@Phantom-42 I have implemented everything that you suggested. Please have a look.
@jayvdb Could you please help me with `!important` ? I am not getting what went wrong here.

blazeu commented:
What happens if you remove important and add `div.` back, does it makes CSSLintBear angry again? is that the reason why there's ignore line before?

Using important is better than having excessive selector IMO. Try adding `/* csslint allow: important */*`

Also, IIRC, StyleLintBear has nicer default than CSSLintBear and CSSLint repo is kind of dead.

blazeu commented:
Okay, I just looked at the default css for this theme, and I found out why we needed `div` or we must use `! important`. The default css has `div` in front of everything. Sorry for catching up late.

In this case using `div.note` is okay, just need to put that csslint ignore line. `/* csslint allow: overqualified-elements */`

kriti21 commented:
@jayvdb @blazeu Please review :)

jayvdb commented:
ack https://github.com/coala/coala/pull/5119/commits/1b94767a1319799d192752565a8e226cef893f00

jayvdb commented:
@gitmate-bot rebase

jayvdb commented:
@gitmate-bot rebase

jayvdb commented:
@gitmate-bot ff
